pandas/core/nanops.py

```diff
@@ -706,11 +706,20 @@ def nanstd(values, axis=None, skipna=True, ddof=1, mask=None):
     >>> nanops.nanstd(s)
     1.0
     """
+    orig_dtype = values.dtype
```

Comment:
Is this fixing `std` behavior in some way? If so do we need a test for this?

Reply:
No, this is a workaround because we have to pass i8 values to nanvar in this case

Reply:
Hmm I think I got it - so previously this would get a junk return value for time delta from the nan call and now we are sending it a mask right? If so is there a reason to even have that call in the first place?

Reply:
> and now we are sending it a mask right

Right.  And this is made necessary because we are passing the values as i8 instead of m8[ns].

> If so is there a reason to even have that call in the first place?

Not sure I understand the question.

Reply:
You can ignore - I was misreading `i8` as an 8 bit integer hence my confusion